### PR TITLE
Update citation component to match refined mocks. #89.

### PR DESCRIPTION
**Pull request by frano-m**
### Reviewers

#89 

@MillenniumFalconMechanic 

### Changes
- created `Citation` component to match refined mocks.
- updated citation stories.
- updated storybook to support the use of alias for component imports.
- replaced all icons to "rounded" in any "touched" (i.e. "finalised" components).
- modified `Link` component for links with copy icon.
- created common `CopyToClipboard` component.
- updated Typography default prop `variant` to `inherit` and removed unnecessary use of this variant.
- css bug fix on customer styled `Paper` component.

### CODE REVIEW

File: explorer/app/components/Project/common/projectTransformer.ts [hunk 1]
```diff
@@ -2,6 +2,19 @@
...
+ * Builds project citation path from projectId.
+ * @param project - Project response model return from API.
+ * @returns string representation of project citation path.
+ */
+export function buildProjectCitationPath(project?: ProjectResponse): string {
+  if (!project) {
+    return "";
+  }
+
+  return `explore/projects/${project.projects[0].projectId}`;
```
Comment (frano-m) [inline explorer/app/components/Project/common/projectTransformer.ts:15]: `explore/projects/` is available in the `SiteConfig` (see `redirectRootToPath`). Let me know if you would like this to be used instead - we can access the value `redirectRootToPath` via the hook `useConfig()`. See line `19` in `Citation` component.

File: explorer/app/components/Project/components/Citation/citation.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+interface Props {
+  citationPath: string;
+}
+
+export const Citation = ({ citationPath }: Props): JSX.Element => {
+  const { datasources } = useConfig();
+  const { url } = datasources;
+  const citationLink = `${url}${citationPath}`;
+  const [root, projectId] = citationPath.split(/\/(?!.*\/)/);
+  const CitationLinkLabel = (
```
Comment (frano-m) [inline explorer/app/components/Project/components/Citation/citation.tsx:23]: CitationLinkLabel is built up to ensure the paths are kept intact as the viewport decreases in size.

File: explorer/app/components/Project/common/projectTransformer.ts (hunk 1 above)
Comment (MillenniumFalconMechanic) [inline explorer/app/components/Project/common/projectTransformer.ts:15]:
```suggestion
  return `explore/projects/${project.projects[0]?.projectId}`;
```

File: explorer/app/components/Project/components/Citation/citation.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+// Styles
+import { CitationLink } from "./citation.styles";
+
+interface Props {
+  citationPath: string;
+}
+
+export const Citation = ({ citationPath }: Props): JSX.Element => {
+  const { datasources } = useConfig();
+  const { url } = datasources;
```
Comment (MillenniumFalconMechanic) [inline explorer/app/components/Project/components/Citation/citation.tsx:20]: It looks like we need to add an additional URL to the config to generate the citation URL. The `{url}` pulled from datasources is the API endpoint but we need the Explorer URL. Take a look at the Angular config to see how we generate it there.

File: explorer/app/components/common/CopyToClipboard/copyToClipboard.tsx [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    setShowTooltip(true);
+  };
+
+  // Timer to auto close the tooltip - the state showTooltip is set to false after a specified time (2 seconds).
+  useEffect(() => {
+    if (showTooltip) {
+      const tooltipTimeout = setTimeout(() => {
+        setShowTooltip(false);
+      }, 2000);
+      return () => clearTimeout(tooltipTimeout);
```
Comment (MillenniumFalconMechanic) [inline explorer/app/components/common/CopyToClipboard/copyToClipboard.tsx:29]: Nice!

**Review by MillenniumFalconMechanic — CHANGES_REQUESTED** [review]
Awesome F, thank you! UX works well too. Just a few nits/q's.

**Pull request merged**